cardinal/ecs/cql/cql.go

```diff
@@ -0,0 +1,135 @@
+package cql
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/participle/v2"
+)
+
+type cqlOperator int
+
+const (
+	opAnd cqlOperator = iota
+	opOr
+)
+
+var operatorMap = map[string]cqlOperator{"&": opAnd, "|": opOr}
+
+func (o *cqlOperator) Capture(s []string) error {
+	*o = operatorMap[s[0]]
```

Comment:
do we need to check that s will have non-zero length?

Reply:
I think the parser lib does not even execute Capture unless it sees a "&" or a "|". But you're right might as well plug this hole.